test/extended/router/external_certificate.go

```diff
@@ -0,0 +1,769 @@
+						resp, err := httpsGetCall(hostName, rootDerBytes)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						o.Expect(resp).Should(o.ContainSubstring(helloOpenShiftResponse))
+					}
+				})
+			})
+
+			g.Context("and the secret is updated but RBAC permissions are dropped", func() {
+				g.BeforeEach(func() {
+					g.By("Deleting RBAC permissions")
+					err = oc.KubeClient().RbacV1().RoleBindings(oc.Namespace()).Delete(context.Background(), secretReaderRoleBinding, metav1.DeleteOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					g.By("Updating the existing secret")
+					// build a new secret
+					secret, rootDerBytes, err = generateTLSCertSecret(oc.Namespace(), "my-tls-secret", corev1.SecretTypeTLS, hosts...)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					// update the existing secret with the new secret
+					_, err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Update(context.Background(), secret, metav1.UpdateOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+				})
+
+				g.It("then routes are not reachable", func() {
+					g.By("Checking the route status")
+					for _, route := range routes {
+						checkRouteStatus(oc, route.Name, corev1.ConditionFalse, "ExternalCertificateValidationFailed")
+					}
+				})
+			})
+
+			g.Context("and the route is updated", func() {
+				var (
+					routeToTest   *routev1.Route
+					newSecretName = "new-ext-crt"
+				)
+
+				g.BeforeEach(func() {
+					// These tests do not *explicitly* need verification on multiple routes.
+					// Hence taking only one route into account.
+					routeToTest = routes[0]
+				})
+
+				g.Context("to use new external certificate", func() {
+					g.BeforeEach(func() {
+						g.By("Creating a new secret")
+						secret, rootDerBytes, err = generateTLSCertSecret(oc.Namespace(), newSecretName, corev1.SecretTypeTLS, hosts...)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						_, err = oc.KubeClient().CoreV1().Secrets(oc.Namespace()).Create(context.Background(), secret, metav1.CreateOptions{})
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						g.By("Updating the existing role to add RBAC permissions for the new secret")
+						err := patchRoleWithSecretAccess(oc, newSecretName)
+						o.Expect(err).NotTo(o.HaveOccurred())
+
+						g.By("Updating the route to use new external certificate")
+						err = patchRouteWithExternalCertificate(oc, routeToTest.Name, newSecretName)
+						o.Expect(err).NotTo(o.HaveOccurred())
+					})
+
+					g.It("then also the route is reachable", func() {
+						g.By("Sending https request")
+						hostName, err := getHostnameForRoute(oc, routeToTest.Name)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						resp, err := httpsGetCall(hostName, rootDerBytes)
+						o.Expect(err).NotTo(o.HaveOccurred())
+						o.Expect(resp).Should(o.ContainSubstring(helloOpenShiftResponse))
+					})
+				})
+
+				g.Context("to use new external certificate, but RBAC permissions are not added", func() {
```

Comment:
What is the point of doing the same tests? That this time a single route is being modified? If so, does this really test a new code path?